### ISSUE TITLE
Running Logos causes ld-linux.so.2 to max out at 100% cpu usage

### ISSUE THREAD

**Issue by aaylnx** [ferion11/LogosLinuxInstaller#15]
Alright after letting Logos run overnight and finish indexing, I'm seeing an issue. While Logos is running fine, a minute or so after it opens up a process called ld-linux.so.2 maxes out to 100% cpu usage when I look at top. When I close Logos, this stops. Anyone else seeing this? Thoughts? I'm using the latest installer running it on a Kubuntu 20.04 install. This does not happen when I use the previously available portable version.

![image](https://user-images.githubusercontent.com/12511481/89053128-5f525e00-d31c-11ea-9bb1-a1ade7a8c0fc.png)

**Comment by ferion11** [comment]
This is the problem with wine 5.x, in fact the process that is consuming this is the `wineserver.exe`.
If you want, you can disable the automatic update of LogosBible, rename the AppImage that is inside the "data" directory and copy the version that is on the old portable version (or download from https://github.com/ferion11/Wine_Appimage/releases the v4.21), which is v4.21 (and use the default v5.11 only for updates). Just replacing the AppImage file is the new way to update wine on the "P" version (you have more control now).
By the way, doing these tests is easier now, as you only need to make a copy of the whole installation directory. And make changes only inside the copy. Keeping the original working independently, regardless of possible failures in modifying the copy.

**Comment by aaylnx** [comment]
Thanks for the reply. Very interesting. Ok, I replaced the 5.11 image with the 4.21 version and you're quite right - no more process issues. My only question is whether or not the 5.x series is ready to be the default for Logos. Seems like the 4.x series is still the better default until this matter gets resolved.  Thanks again for all of your work Daniel!!!

**Comment by ferion11** [comment]
Just to add one more note:
* You can change one installation made with AppImage to use native Wine by just renaming the `bin` directory inside the `data` directory (to something like `bin_backup`). If it (`Logos.sh` or `controlPanel.sh`) can find the `data/bin` then it will use it, else it will use the one from your system.

* Now, from native Wine32 installation to AppImage (in case of not liking the Linux distribution default), you just need the AppImage and the `bin` directory (for the v4.21 wine, or you can get any version you want or have downloaded too):
```
$ cd LogosBible_Linux_P/data
$ wget -c https://github.com/ferion11/Wine_Appimage/releases/download/v4.21/wine-i386_x86_64-archlinux.AppImage
$ mkdir bin
$ cd bin
$ ln -s ../wine-i386_x86_64-archlinux.AppImage wine
$ ln -s ../wine-i386_x86_64-archlinux.AppImage wineserver
```

* On the native Wine64 installation, you will have to change the Wine version on the Linux distribution.

**Comment by ferion11** [comment]
> Thanks for the reply. Very interesting. Ok, I replaced the 5.11 image with the 4.21 version and you're quite right - no more process issues. My only question is whether or not the 5.x series is ready to be the default for Logos. Seems like the 4.x series is still the better default until this matter gets resolved. Thanks again for all of your work Daniel!!!

The problem here is that the LogosBible v8.15 have from https://wiki.logos.com/Logos_8.15

> Known Issue for Windows: When updating to Logos 8.15 some users will encounter a “Smart Screen” notification on Windows. This is because we renewed our code-signing certificate. Please click “Allow It” and the application install will continue and install safely. We expect this issue to go away as more customers download and run the installer.

Even on Windows this can cause problems, and, because of this, installing or updating with v4.21 causes the related error (and not working):
```
0024:fixme:ntdll:NtLockFile I/O completion on lock not implemented yet
```
But this problem is only with the installer/updater.

**Comment by aaylnx** [comment]
Very interesting. This must explain the errors that I was getting (but could simply click through without a problem) using the portable version. Since I've installed with the 5.11 version, but now replaced it with 4.21 I now have no error pop-ups and no cpu usage. The best of both worlds, though it was a little work. Thanks very much for the info!

**Comment by ferion11** [comment]
> Very interesting. This must explain the errors that I was getting (but could simply click through without a problem) using the portable version. Since I've installed with the 5.11 version, but now replaced it with 4.21 I now have no error pop-ups and no cpu usage. The best of both worlds, though it was a little work. Thanks very much for the info!

I am planning a way to compile PlayOnLinux-Wine with custom patch, using automated procedures inside the travis-ci (because my system is no-multilib, pure 64bits, making hard to compile Wine from the code without having 32bits support, and that's the main reason I use AppImage). So I can start to fix these issues directly in the Wine code, along with other Wine developers.
Thanks you too :)

By the way, the LogosBible 64bits need the 64bits Wine64 plus the 32bits Wine32 on the WoW64 installation (it double the disk space needed), so pure 64bits Wine64 can't run LogosBible 64bits either.
WoW64 - from https://wiki.winehq.org/Building_Wine - When Windows began targeting 64-bit architectures, Microsoft decided to include a compatibility layer to support their massive universe of 32-bit applications. This kind of subcomponent, nicknamed WoW64 (for Windows on Windows 64-bit), is also implemented in Wine to solve the exact same problem.

**Comment by ferion11** [comment]
Just to add some more info note here (so don't worry): maybe using Wine esync patchset would be a good solution for thus issue, because it avoid using wineserver.exe, maybe even one good speed improvement too. But at this day, the basic patch is just for v5.7 or lower (waiting for the patch update first, to start testing).

**Comment by ferion11** [comment]
Fixed with the AppImage of [v2.18](https://github.com/ferion11/LogosLinuxInstaller/releases), **or** using the native wine v5.18 or higher.
